packages/react-instantsearch-core-next/src/__tests__/InstantSearch.test.tsx

```diff
@@ -0,0 +1,129 @@
+import { render } from '@testing-library/react';
+import React, { version as ReactVersion } from 'react';
+
+import { createSearchClient } from '../../../../test/mock';
+import { IndexContext } from '../IndexContext';
+import { InstantSearch } from '../InstantSearch';
+import { InstantSearchContext } from '../InstantSearchContext';
+import version from '../version';
+
+import type { InstantSearch as InstantSearchType } from 'instantsearch.js';
+import type { IndexWidget } from 'instantsearch.js/es/widgets/index/index';
+
+describe('InstantSearch', () => {
+  test('renders children', () => {
+    const searchClient = createSearchClient();
+
+    const { container } = render(
+      <InstantSearch indexName="indexName" searchClient={searchClient}>
+        Children
+      </InstantSearch>
+    );
+
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        Children
+      </div>
+    `);
+  });
+
+  test('provides the search instance', () => {
+    const searchClient = createSearchClient();
+    let searchContext: InstantSearchType | null = null;
+
+    render(
+      <InstantSearch indexName="indexName" searchClient={searchClient}>
+        <InstantSearchContext.Consumer>
+          {(context) => {
+            searchContext = context;
+            return null;
+          }}
+        </InstantSearchContext.Consumer>
+      </InstantSearch>
+    );
+
+    expect(searchContext).toEqual(
+      expect.objectContaining({
+        start: expect.any(Function),
+        dispose: expect.any(Function),
+        addWidgets: expect.any(Function),
+        removeWidgets: expect.any(Function),
+      })
+    );
+  });
+
+  test('provides the main index', () => {
+    const searchClient = createSearchClient();
+    let indexContext: IndexWidget | null = null;
+
+    render(
+      <InstantSearch indexName="indexName" searchClient={searchClient}>
+        <IndexContext.Consumer>
+          {(context) => {
+            indexContext = context;
+            return null;
+          }}
+        </IndexContext.Consumer>
+      </InstantSearch>
+    );
+
+    expect(indexContext).toEqual(
+      expect.objectContaining({
+        $$type: 'ais.index',
+        addWidgets: expect.any(Function),
+        removeWidgets: expect.any(Function),
+      })
+    );
+  });
+
+  test('attaches users agents', () => {
+    const searchClient = createSearchClient();
+
+    render(<InstantSearch indexName="indexName" searchClient={searchClient} />);
+
```

Comment:
```suggestion
    expect(searchClient.addAlgoliaAgent).toHaveBeenCalledTimes(2);
```

or is it called more times?

Reply:
For now it's called more because props are not stable (handled in the next PR!).

Reply:
Added in TODO in fa71236b725a27cba12e4588a31a3adf57676aba.